packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -792,44 +794,47 @@ class _ModalScopeState<T> extends State<_ModalScope<T>> {
             bucket: widget.route._storageBucket, // immutable
             child: Actions(
               actions: _actionMap,
-              child: FocusScope(
-                node: focusScopeNode, // immutable
-                child: RepaintBoundary(
-                  child: AnimatedBuilder(
-                    animation: _listenable, // immutable
-                    builder: (BuildContext context, Widget? child) {
-                      return widget.route.buildTransitions(
-                        context,
-                        widget.route.animation!,
-                        widget.route.secondaryAnimation!,
-                        // This additional AnimatedBuilder is include because if the
-                        // value of the userGestureInProgressNotifier changes, it's
-                        // only necessary to rebuild the IgnorePointer widget and set
-                        // the focus node's ability to focus.
-                        AnimatedBuilder(
-                          animation: widget.route.navigator?.userGestureInProgressNotifier ?? ValueNotifier<bool>(false),
-                          builder: (BuildContext context, Widget? child) {
-                            final bool ignoreEvents = _shouldIgnoreFocusRequest;
-                            focusScopeNode.canRequestFocus = !ignoreEvents;
-                            return IgnorePointer(
-                              ignoring: ignoreEvents,
-                              child: child,
+              child: PrimaryScrollController(
+                controller: ScrollController(),
```

Comment:
We should probably not instantiate a new controller every time this rebuilds and instead reuse the same scrollcontroller instance?

Reply:
Fixed!